fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/Column.java

```diff
@@ -77,6 +78,10 @@ public static <V extends Value> Column<V> unnamedOf(@Nonnull final V value) {
         return new Column<>(Field.unnamedOf(value.getResultType()), value);
     }
 
+    public static <V extends Value> Column<V> of(@Nonnull Optional<String> fieldNameOptional, @Nonnull final V value) {
+        return new Column<>(Field.of(value.getResultType(), fieldNameOptional), value);
+    }
```

Comment:
This new helper method allows us clean up code, because there were a fair number of cases where we took the result type from the `value`, and then wanted to attach a field name. By line count, most of the changes in this PR are mechanical updates to use this new helper method where appropriate

Reply:
The benefit of using an `Optional` here is that in a case where you're starting with an `Optional` field name, you can call this method instead of doing something like:

```java
Column<V> column = maybeFieldName
    .map(fieldName -> Column.of(fieldName, value))
    .orElseGet(() -> Column.unnamedOf(value))
```

Instead, you can do:

```java
Column<V> column = Column.of(maybeFieldName, value);
```

I think there was a version of this PR where I actually made use of this to do something like `Column.of(field.getFieldNameOptional(), value)` when copying over the value from one field into another. That has since been removed, so it's a little less obvious.

Note that we end up wrapping the field name in an optional when creating the `Field` anyway, so it's not like we can avoid an object creation by having a string version.

I could see the benefit to _also_ having a variation that takes a `String`, to avoid needing quite as many `Optional.of`s at call sites.